src/renderers/shared/reconciler/ReactStatelessComponent.js

```diff
@@ -0,0 +1,396 @@
+/**
```

Comment:
I'm assuming this is (mostly) just a clone of ReactCompositeComponent?  Is there a reason we can't do something simple like call createClass and thereby make it into a full CompositeComponent?